Index: usbmode-2017-12-19-f40f84c2/usb-modeswitch-data-20191128/usb_modeswitch.d/413c:81d7
===================================================================
--- usbmode-2017-12-19-f40f84c2.orig/usb-modeswitch-data-20191128/usb_modeswitch.d/413c:81d7
+++ usbmode-2017-12-19-f40f84c2/usb-modeswitch-data-20191128/usb_modeswitch.d/413c:81d7
@@ -1,2 +1,3 @@
 # Dell DW5821e
-Configuration=0
+Configuration=1
+NoDriverLoading=1
